packages/api/src/webhooks/cannon.ts

```diff
@@ -402,36 +407,84 @@ export default class WebhookCannon {
         );
         // we don't retry on non 400 responses. only on timeouts
         // this.retry(event);
-        return;
       } catch (e) {
         console.log("firing error", e);
+        errorMessage = e.message;
         await this.retry(trigger, params, e);
+      } finally {
+        await this.storeResponse(webhook, event, resp, startTime, responseBody);
+        await this.storeTriggerStatus(
+          trigger.webhook,
+          triggerTime,
+          statusCode,
+          errorMessage
+        );
         return;
       }
     }
   }
 
+  async storeTriggerStatus(
+    webhook: DBWebhook,
+    triggerTime: number,
+    statusCode?: number,
+    errorMessage?: string
+  ) {
+    try {
+      if (statusCode >= 300 || !statusCode) {
+        await this.db.webhook.updateStatus(webhook.id, {
+          status: {
+            lastTriggeredAt: triggerTime,
+            lastFailure: {
+              timestamp: triggerTime,
+              statusCode: statusCode,
+              errorMessage: errorMessage,
+            },
+          },
+        });
+      } else {
+        await this.db.webhook.updateStatus(webhook.id, {
+          status: {
+            ...webhook.status,
+            lastTriggeredAt: triggerTime,
+          },
+        });
```

Comment:
**nit:** A more linear implementation calling `updateStatus` only in 1 place, wdyt?
```suggestion
      let status: DBWebhook["status"] = { lastTriggeredAt: triggerTime };
      if (statusCode >= 300 || !statusCode) {
        status = {
          ...status, 
          lastFailure: {
            timestamp: triggerTime,
            statusCode,
            errorMessage,
          }
        };
      }
      await this.db.webhook.updateStatus(webhook.id, status);
```